File: MyPython/ming/LaunchGooglePlay.py
'''
Created on 2014/3/31

@author: YuMing
'''
from uiautomator import device as d

if __name__ == '__main__':

    d.press.home()

    view = d(className='android.view.View', index=4)

    all_apps_button = view.child(className='android.widget.TextView', index=2)
    all_apps_button.click()

    google_play_icon = d(description='Shop')
    google_play_icon.click()

    play_store_text = d(text='Play Store')
    assert play_store_text.wait.exists(timeout=5000), 'Play Store Text does not exist.'

File: Uiautomator/src/com/uia/example/my/LaunchGooglePlay.java
package com.uia.example.my;

import com.android.uiautomator.core.UiObject;
import com.android.uiautomator.core.UiObjectNotFoundException;
import com.android.uiautomator.core.UiSelector;
import com.android.uiautomator.testrunner.UiAutomatorTestCase;

public class LaunchGooglePlay extends UiAutomatorTestCase {

	public void testDemo() throws UiObjectNotFoundException {

		getUiDevice().pressHome();
		
		UiObject allAppsButton = new UiObject(new UiSelector().className(android.view.View.class).index(4))
			.getChild(new UiSelector().className(android.widget.TextView.class).index(2));
		allAppsButton.click();

		UiObject googlePlayIcon = new UiObject(new UiSelector().description("Shop"));
		googlePlayIcon.click();

		UiObject playStoreText = new UiObject(new UiSelector().text("Play Store"));
		assertTrue("Play Store Text does not exist.", playStoreText.waitForExists(5000));
	}
}